test/FunctionalTests/Microsoft.OData.Core.Tests/AsyncOnlyStreamWrapper.cs

```diff
@@ -0,0 +1,185 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="AsyncOnlyStreamWrapper.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.OData.Tests
+{
+    /// <summary>
+    /// A test stream wrapper that that throws an
+    /// exception when synchronous I/O methods are called.
+    /// This is meant for testing asynchronous code paths
+    /// where synchronous I/O is not allowed.
+    /// </summary>
+    internal class AsyncOnlyStreamWrapper : Stream

```

Comment:
```suggestion
    internal class AsyncStream : Stream
```
`AsyncOnlyStreamWrapper` is quite a mouthful. That it's a wrapper is covered in the doc string.